components/screens/MenuScreen.tsx

```diff
@@ -1,32 +1,90 @@
-import {useContext} from 'react';
+import React from 'react';
 
 import {StyleSheet} from 'react-native';
 import {SafeAreaView} from 'react-native-safe-area-context';
 
+import * as Updates from 'expo-updates';
+
 import {Heading, HStack, VStack, Switch, Text} from 'native-base';
 
-import {ClientContext, ClientProps} from 'clientContext';
-import {AvalancheCenterSelector} from 'components/AvalancheCenterSelector';
+import {AvalancheCenterCard, AvalancheCenterSelector} from 'components/AvalancheCenterSelector';
+
+import {createNativeStackNavigator, NativeStackScreenProps} from '@react-navigation/native-stack';
 
-export const MenuScreen = () => {
-  const {staging, toggleStaging, setAvalancheCenter} = useContext<ClientProps>(ClientContext);
+import {MenuStackParamList, MenuStackNavigationProps} from 'routes';
+import {useNavigation} from '@react-navigation/native';
+
+const MenuStack = createNativeStackNavigator<MenuStackParamList>();
+export const MenuStackScreen = (
+  center_id: string,
+  setAvalancheCenter: React.Dispatch<React.SetStateAction<string>>,
+  staging: boolean,
+  setStaging: React.Dispatch<React.SetStateAction<boolean>>,
+) => {
   return (
-    <SafeAreaView style={styles.fullscreen}>
-      <VStack pt="16" px="4" space="4" style={styles.fullscreen}>
-        <Heading>Settings</Heading>
-        <Heading size="sm">Debug settings</Heading>
-        <HStack justifyContent="space-between" alignItems="center" space="4">
-          <Text>Use staging environment</Text>
-          <Switch value={staging} onValueChange={toggleStaging} />
-        </HStack>
-        <AvalancheCenterSelector setAvalancheCenter={setAvalancheCenter} />
-      </VStack>
-    </SafeAreaView>
+    <MenuStack.Navigator initialRouteName="menu">
+      <MenuStack.Screen name="menu" component={MenuScreen(staging, setStaging)} initialParams={{current_center_id: center_id}} options={{title: `Settings`}} />
+      <MenuStack.Screen
+        name="avalancheCenterSelector"
+        component={AvalancheCenterSelectorScreen(setAvalancheCenter)}
+        initialParams={{current_center_id: center_id}}
+        options={{title: `Choose An Avalanche Center`}}
+      />
+    </MenuStack.Navigator>
```

Comment:
this feels like a lot of code to essentially allow someone to pick from a list. do you really need a separate screen and the navigation mechanism?

to be clear, I don't feel that strongly about it - it's not something we're shipping! but it does make this source file a bit more busy than it was before.

Reply:
Hard to say what is a hard need, but I was poking around the other apps on my phone and in pretty much every case, the top-level menu screen had generic-sounding options, each of which would navigate you to a new screen where you made decisions - so I figured aside from learning the navigation bits for myself, if/when we actually have things to be setting in the menu, we would likely end up with a similar pattern?

Reply:
True, the UX pattern is common! 

I did spot https://github.com/jsoendermann/react-native-settings-screen while looking around last week and feel like something like that might be a decent option to explore if we end up with more settings to add.

Reply:
Cool idea - that particular library hasn't seen a commit in 4 years, though - do you think that it's safe to use?